src/recipients/recipient.ts

```diff
@@ -174,26 +174,29 @@ export function getRecipientFromAddress(
 
 export function getRecipientVerificationStatus(
   recipient: Recipient,
-  e164NumberToAddress: E164NumberToAddressType
+  e164NumberToAddress: E164NumberToAddressType,
+  verifiedAddresses: string[]
 ): RecipientVerificationStatus {
-  if (recipientHasAddress(recipient)) {
```

Comment:
this logic needed some rejiggering - if we just replaced this top check with e.g.

```
if (recipientHasAddress(recipient)) {
      return verifiedAddresses.includes(recipient.address)
        ? RecipientVerificationStatus.VERIFIED
        : RecipientVerificationStatus.UNVERIFIED
}
```

then we'd potentially return `UNVERIFIED` for phone recipients who happen to have an address. this new layout gives preference to phone numbers; e.g., if a recipient has both an address and a phone number, we do the phone number checks; we only do the address checks if the recipient _only_ has an address.